src/parquet/util/CMakeLists.txt

```diff
@@ -40,6 +41,7 @@ add_library(parquet_util STATIC
   cpu-info.cc
   input.cc
   mem-pool.cc
+  memory-pool.cc
```

Comment:
Perhaps we could refactor the Impala `mem-pool.cc` (whose API is different from the simpler Allocate/Free memory pool) to be called `ChunkedMemPool` or something and use the DefaultMemoryPool for its chunk malloc/free calls? We could also nix the chunked pool altogether in favor of quadratically-growing allocations in `DictEncoder`